pkg/validate/validate.go

```diff
@@ -32,19 +33,108 @@ func Validate(mb utils.MetaBundle) (bool, []error) {
 	return allSuccess, errs
 }
 
-func GetAllValidators() []utils.Validator {
-	return []utils.Validator{
-		{
-			Description: "Ensure defaultChannel is present in list of channels",
-			Runner:      validators.ValidateDefaultChannel,
-		},
-		{
-			Description: "Ensure `label` to follow the format api.openshift.com/addon-<operator-id>",
-			Runner:      validators.ValidateAddonLabel,
-		},
-		{
-			Description: "Ensure current csv is present in the index image",
-			Runner:      validators.ValidateCSVPresent,
-		},
+// name formatting rule: [0-9]{3}_([a-z]*_?)*
+var AllValidators = map[string]utils.Validator{
+	"001_default_channel": {
+		Description: "Ensure defaultChannel is present in list of channels",
+		Runner:      validators.ValidateDefaultChannel,
+	},
+	"002_label_format": {
+		Description: "Ensure `label` to follow the format api.openshift.com/addon-<operator-id>",
+		Runner:      validators.ValidateAddonLabel,
+	},
+	"003_csv_present": {
+		Description: "Ensure current csv is present in the index image",
+		Runner:      validators.ValidateCSVPresent,
+	},
+}
+
+func GetExistingValidatorNames() string {
+	var res []string
+	for name := range AllValidators {
+		res = append(res, name)
+	}
+	return strings.Join(res, ",")
+}
+
+type validatorsFilter struct {
+	ReturnAll      bool
+	Enabled        bool
```

Comment:
Do we really need to track `Enabled`? :/
can't it be just 
```go
type validatorsFilter struct {
    ReturnAll bool // same meaning as before
    validatorNames []string // effective names of validators to finally consider (enabled validators / all minus disabled validators)
}
```

And the `NewFilter` function/constructor can have the following pseudocode
```
if disabled == "" and enabled == "" 
    return ValidatorsFilter{ReturnAll: true}

if enabled != "" 
    validatorNames := strings.Split(enabled, ",")
if disabled != ""
    validatorNames := strings.Split(allValidators minus disabled, ",")

return validatorsFilter {ReturnAll: false, validatorsNames: validatorNames}
```

With this we can easily modify the `GetValidators` function to this
```
output := []utils.Validator
if f.returnAll {
    for _, validator := range AllValidators {
        output = append(output, validator)    ​
    }
} else {
    for _, name := f.validatorNames {
        validator := AllValidators[name] // no need to track 'ok' as it was already validated in the NewFilter function
        output = append(output, validator)
    }
}

return output
```

With this, codebase would become much cleaner as `GetAllValidators()`, `GetEnabledValidators`, `GetDisabledValidators` would have gotten rid of.

Looks pretty simple and cleaner to me, wdyt :)

Reply:
I like this, let me refactor my code.